### DISCUSSION TITLE
RU Authentication Failure (Credential missing) 401

### DISCUSSION THREAD

**Discussion by ruiyan-yu** [Digitaler-Impfnachweis/certification-apis#147] [Q&A] [answered]
Hallo,

Der Fehler tritt seit heute morgen auf.
Authentifizierung wird via TI-Konnektor duchgeführt.

Mit freundlichen Grüßen

Duria e.G

**Comment by msteffenDev** [comment]
Wir erhalten den Fehler auch seit gestern, was kann die Ursache sein, gibt es technische Probleme in der RU ?

*   Trying 10.30.17.10...
* TCP_NODELAY set
* Connected to id.ru.impfnachweis.info (10.30.17.10) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.2 (IN), TLS handshake, Server hello (2):
{ [64 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [4727 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [657 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [138 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
{ [1 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: 1.3.6.1.4.1.311.60.2.1.3=DE; 1.3.6.1.4.1.311.60.2.1.2=Nordrhein-Westfalen; 1.3.6.1.4.1.311.60.2.1.1=Koeln; businessCategory=Private Organization; serialNumber=HRB 82994; C=DE; ST=Nordrhein-Westfalen; L=Koeln; postalCode=50670; street=Im Mediapark 5; O=ubirch GmbH; CN=api.ru.impfnachweis.info
*  start date: May 10 14:15:13 2021 GMT
*  expire date: May 13 14:15:13 2022 GMT
*  issuer: C=DE; O=D-Trust GmbH; CN=D-TRUST SSL Class 3 CA 1 EV 2009
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x7ff5ee00d200)
> GET /auth/realms/ubirch-certify/login-actions/authenticate?session_code=1jmmqLpX0EFXdAxhNAbBuS5FcrhQ37etDVUsLMr6Pz8&execution=fac97347-d739-43a5-adeb-1f600009e0ba&client_id=user-access-ti&tab_id=V3eqN0fTy2I&auth_session_id=659492f6-84fb-4624-bc30-4c7e9b31cc59 HTTP/2
> Host: id.ru.impfnachweis.info
> User-Agent: curl/7.54.0
> Accept: */*
> x-auth-signed-challenge: R7uoaeRr0rbe4ca0kkdjNNe1iSeKMaJiOV/VShzoM4cASTpdOZAMMCZ/7lDT0lpCSutU8d+iEQwrN79XL64bNqnAgZxwiiT7HzWhYs/ARLai9cZe2jzzZyHZFVt55j0Y8en6rqwKlaQcz0bnZEFE7iEYtmaMFGSVEq6oJYmH0NWJw9t0NLvTml3AezzKvUrsB1pNO4E/Kj4pgeIfueCActvI5YitE7cpyfThDY3l8lYEcvrR/afX1ao8bKGClD7/g1219bO37crZY5qyK6AccuqleIkr/qgd5bM3O1P8UyhojmT6pek4FYEkTvQ1oJhGCaoH7J38HLNvrfKiPGIWLQ==
> x-auth-certificate: MIIFujCCBKKgAwIBAgIEdzWU0TANBgkqhkiG9w0BAQsFADCBnDELMAkGA1UEBhMCREUxIDAeBgNVBAoMF21lZGlzaWduIEdtYkggTk9ULVZBTElEMUgwRgYDVQQLDD9JbnN0aXR1dGlvbiBkZXMgR2VzdW5kaGVpdHN3ZXNlbnMtQ0EgZGVyIFRlbGVtYXRpa2luZnJhc3RydWt0dXIxITAfBgNVBAMMGE1FU0lHLlNNQ0ItQ0ExIFRFU1QtT05MWTAeFw0xODExMjExMTM3NDZaFw0yMzExMjExMTM3NDZaMIGDMQswCQYDVQQGEwJERTEoMCYGA1UECgwfMi0yLjM3LjIuMS5SMDU3MzYxNjAwMDAwMDA0Nzk3NTEgMB4GA1UEBRMXMTYuODAyNzYwMDI3OTE2OTk5MDAxMTExKDAmBgNVBAMMH0RyLi9NZWQuKEgpIEhlw7x2ZWxkb3BfIE9lci1FcmswggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCUuSLd8UApM3nReG+VYksVAUQOYuf/firyA4K0XhrFRkeMnTEiWeWDXnTZxCYXbYYQWjOCzlenEFnJVi8D74n6JmrBARNaVuMs1LagiWkTjFv39tficY4MX8M2/6+K1C67FiWwfn2jX0Y4cOQ/RldP99G7td9DMUc+MFBiEs0L05+G2rLjWXuNSd4CZ+OI5fTMvRCIgRnByYuVcgaQl9+79bZETutPqMFhzI6jsE9XKQOlYLT1lkKKuv3i0b3xf04rXdPI7AcuF5edSbq4yq3qut26ufO7anKjWySLeVbRx7W+Afh0PoPnwfFEu+9molXvO5WYIAtbfLXl4xNkPJipAgMBAAGjggIZMIICFTAMBgNVHRMBAf8EAjAAMB8GA1UdIwQYMBaAFM/7L1sRv+tAcf18el5zKtTEKuEDMFYGCCsGAQUFBwEBBEowSDBGBggrBgEFBQcwAYY6aHR0cDovL2ZvY3NwLXNtY2ItYXV0LnRlc3R1bWdlYnVuZy5kZ25zZXJ2aWNlLmRlOjgwODAvb2NzcDAgBgNVHSAEGTAXMAoGCCqCFABMBIEjMAkGByqCFABMBE0wEwYDVR0lBAwwCgYIKwYBBQUHAwIwgaIGA1UdHwSBmjCBlzCBlKCBkaCBjoaBi2xkYXA6Ly9sZGFwLnRlc3R1bWdlYnVuZy5kZ25zZXJ2aWNlLmRlOjM4OS9DTj1DUkwtMTMsTz1tZWRpc2lnbiUyMEdtYkgsQz1ERT9jZXJ0aWZpY2F0ZVJldm9jYXRpb25MaXN0P2Jhc2U/b2JqZWN0Q2xhc3M9Y1JMRGlzdHJpYnV0aW9uUG9pbnQwHQYDVR0OBBYEFFEv/UZP1zwp4AU8RqXwMinZ7ZqaMA4GA1UdDwEB/wQEAwIFoDCBgAYFKyQIAwMEdzB1pC0wKzEcMBoGA1UECgwTS1pWIFdlc3RmYWxlbi1MaXBwZTELMAkGA1UEBhMCREUwRDBCMEAwPjAQDA5aYWhuYXJ6dHByYXhpczAJBgcqghQATAQzEx8yLTIuMzcuMi4xLlIwNTczNjE2MDAwMDAwMDQ3OTc1MA0GCSqGSIb3DQEBCwUAA4IBAQCOz89BISfYBRxLkRLpggYVgN8IE24g0hKrR5B0foF1rs38i3htIwPT3ns1jbhQ3vTES6JbrKIxlL08UXr9c6xxaM1T95DmdqnZn70VhUAkMcEgAvTJX/ebTS09Z4Rv4oj87O9vHlJYP15oqKnGeCLG8K1QsT8JlNULS/ICGTCywn59HCSEpC8da8ZfgF9xDX8uw/+hp/lqmwfVw65Eb9In98yM32WzQ2GZBxVNeqW8vh3AVMEJA/guiAs5Ss8bL380PZIi8cJk6WAWDGJwYeg5XCby2Z5WQNT1BcAuGyUii+AtsdzCZqBS5bHjVHa+0hRbaf7/teEXk6G4XTYqSaQi
> 
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 401 
< cache-control: no-store, must-revalidate, max-age=0
< content-language: en
< content-security-policy: frame-src 'self'; frame-ancestors 'self'; object-src 'none';
< content-type: text/html;charset=utf-8
< date: Tue, 29 Jun 2021 07:15:43 GMT
< referrer-policy: no-referrer
< server: istio-envoy
< strict-transport-security: max-age=15552000; includeSubDomains; preload
< x-content-type-options: nosniff
< x-envoy-decorator-operation: ruproxy.ubirch-prod.svc.cluster.local:8080/*
< x-envoy-upstream-service-time: 30
< x-frame-options: SAMEORIGIN
< x-robots-tag: none
< x-xss-protection: 1; mode=block
< content-length: 2418
< 
{ [2418 bytes data]
* Connection #0 to host id.ru.impfnachweis.info left intact
HTTP/2 401 
cache-control: no-store, must-revalidate, max-age=0
content-language: en
content-security-policy: frame-src 'self'; frame-ancestors 'self'; object-src 'none';
content-type: text/html;charset=utf-8
date: Tue, 29 Jun 2021 07:15:43 GMT
referrer-policy: no-referrer
server: istio-envoy
strict-transport-security: max-age=15552000; includeSubDomains; preload
x-content-type-options: nosniff
x-envoy-decorator-operation: ruproxy.ubirch-prod.svc.cluster.local:8080/*
x-envoy-upstream-service-time: 30
x-frame-options: SAMEORIGIN
x-robots-tag: none
x-xss-protection: 1; mode=block
content-length: 2418

    <!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"
            \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">
    <html xmlns=\"http://www.w3.org/1999/xhtml\" class=\"login-pf\">

    <head>
        <meta charset=\"utf-8\">
        <meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\"/>
        <meta name=\"robots\" content=\"noindex, nofollow\">

                <meta name=\"viewport\" content=\"width=device-width,initial-scale=1\"/>
        <title>Log in to Ubirch Certify App Login</title>
        <link rel=\"icon\" href=\"/auth/resources/mrpuj/login/Ubirch-simple/img/favicon.ico\"/>
                <link href=\"/auth/resources/mrpuj/common/keycloak/node_modules/patternfly/dist/css/patternfly.min.css\" rel=\"stylesheet\"/>
                <link href=\"/auth/resources/mrpuj/common/keycloak/node_modules/patternfly/dist/css/patternfly-additions.min.css\" rel=\"stylesheet\"/>
                <link href=\"/auth/resources/mrpuj/common/keycloak/lib/zocial/zocial.css\" rel=\"stylesheet\"/>
                <link href=\"/auth/resources/mrpuj/login/Ubirch-simple/css/login-28042021.css\" rel=\"stylesheet\"/>
    <SCRIPT> if (typeof history.replaceState === 'function') {  history.replaceState({}, \"some title\", \"https://id.ru.impfnachweis.info/auth/realms/ubirch-certify/login-actions/authenticate?execution=fac97347-d739-43a5-adeb-1f600009e0ba&client_id=user-access-ti&tab_id=V3eqN0fTy2I\"); }</SCRIPT></head>

    <body class=\"\">

    <div class=\"header\">
        <a class=\"logo\" href=\"https://ubirch.de\">
            <img src=\"/auth/resources/mrpuj/login/Ubirch-simple/img/Ubirch_Logo_weiss.svg\" alt=\"logo\"></a>
        <div id=\"kc-header-wrapper\"
             class=\"\">Ubirch Certify App Login</div>
    </div>

    <div class=\"login-pf-page\">
        <div id=\"kc-header\" class=\"login-pf-page-header\">
            <div id=\"kc-header-wrapper\"
                 class=\"\">Ubirch Certify App Login</div>
        </div>
        <div class=\"card-pf \">
            <header class=\"login-pf-header\">
                        <h1 id=\"kc-page-title\">        We are sorry...
</h1>
            </header>
            <div id=\"kc-content\">
                <div id=\"kc-content-wrapper\">


        <div id=\"kc-error-message\">
            <p class=\"instruction\">Certificate & Signature Verification failed</p>
        </div>


                </div>
            </div>

        </div>
    </div>
    </body>
    </html>

**Comment by MirkoWe** [comment]
Ich schließe mich hier an. Am Freitag war die TI RU gar nicht erreichbar und seit gestern bekommen wir den gleichen Fehler wie @msteffenDev.

**Reply by norganim** [reply]
Dito "Certificate & Signature Verification failed"

**Reply by alexanderwilms** [reply]
Same here: "Certificate & Signature Verification failed"

**Reply by marcusthierfelder** [reply]
Bei uns auch

**Reply by dieAte** [reply]
Bitte versucht über den Realm bmg-ti-certify zuzugreifen statt über ubirch-certify

**Reply by marcusthierfelder** [reply]
Vielen Dank, hilft aber nicht

**Reply by norganim** [reply]
we tested both TI RU endpoints (new and deprecated), result is always the same "Certificate & Signature Verification failed"

**Comment by alexanderwilms** [comment]
@oliver-steinbrecher - irgendeine Idee, wann dieses Problem behoben wird?  Wir brauchen die RU-Umgebung dringend für Tests und Zertifizierung!

**Comment by oliver-steinbrecher** [comment, marked as answer]
Update deployed - please retest

**Reply by msteffenDev** [reply]
getestet, funktioniert

**Reply by oliver-steinbrecher** [reply]
Danke für die Rückmeldung

**Reply by ruiyan-yu** [reply]
Danke, es funktioniert wieder. Was wurde denn aktualisiert? Genesenen-Impfung scheint es nicht zu sein.

**Reply by dieAte** [reply]
Das TSL Zertifikat des SMCB Connectors
